### DISCUSSION TITLE
Unpredictable test warnings

### DISCUSSION THREAD

**Discussion by Tezei** [Cyfrin/foundry-full-course-cu#2752] [General]
Hiya guys,

During test steps I am receiving test warnings as follows:

Warning (2018): Function state mutability can be restricted to view
  --> test/FundMeTest.t.sol:30:5:
   |
30 |     function testMinimumUSDIsFive() public {
   |     ^ (Relevant source part starts here and spans across multiple lines).

Warning (2018): Function state mutability can be restricted to view
  --> test/FundMeTest.t.sol:35:5:
   |
35 |     function testOwnerIsMsgSender() public {
   |     ^ (Relevant source part starts here and spans across multiple lines).

Warning (2018): Function state mutability can be restricted to view
  --> test/FundMeTest.t.sol:39:5:
   |
39 |     function testPriceFeedVersionIsAccurate() public {
   |     ^ (Relevant source part starts here and spans across multiple lines).
   
   I redeclare functions as views to run the tests and warnings are gone. Should I stay with view modifier?
   
   Test functions are note defined as view in the lesson. Please advise.
   
   10x in advance!

**Comment by EngrPips** [comment]
Hello @Tezei, You can leave the `view` keyword, and that is fine, but if you want to remove it as the course codebase doesn't have you can remove it, and everything should still work fine, but only with the warnings you are getting which is not a problem at all.

**Reply by Tezei** [reply]
10x a lot!

**Reply by EngrPips** [reply]
Anytime.

**Comment by DCVglobalnetwork** [comment]
You can also use the Solidity Extension in your code editor, so before running in the terminal, it will show you an error.

**Reply by Tezei** [reply]
Which one, I am using this recommended in the course?

**Reply by DCVglobalnetwork** [reply]
I got a few; I do not remember which one was exactly in the course 
I think you are referring to the one below.

![image](https://github.com/user-attachments/assets/19330783-ab63-4d25-875e-228bff749465)

I have also 

![image](https://github.com/user-attachments/assets/799bfac8-22c6-43dc-a69b-a0a2a2b72fe4)

![image](https://github.com/user-attachments/assets/20553264-33d1-42f6-ab9e-cebdb5d2fd52)

**Reply by EngrPips** [reply]
fair enough

**Comment by EngrPips** [comment]
Indeed

**Comment by Tezei** [comment]
10x a lot!

**Reply by EngrPips** [reply]
Anytime
